viite-backend/viite-main/src/test/scala/fi/liikennevirasto/viite/process/DefaultSectionCalculatorStrategySpec.scala

```diff
@@ -2940,4 +2940,167 @@ Left     ^  ^   Right
+        LinearLocation(465419,8.0,  "be5499ac-03bf-4fca-abac-bd8af0e7eaf3:2",0.0,13.135,    SideCode.TowardsDigitizing,1698414053000L,(CalibrationPointReference(Some(206),Some(JunctionPointCP)),CalibrationPointReference(Some(219),Some(JunctionPointCP))),List(Point(443494.906,7375886.053,0.0), Point(443501.416,7375897.461,0.0)),FrozenLinkInterface,roadwayNumberLeftTrackSection,Some(DateTime.now().minusDays(2)),None),
+        LinearLocation(465377,8.0,  "9acdcdf3-7a11-49e8-9f5f-f62863290f5a:2",0.0,13.658,    SideCode.TowardsDigitizing,1698414053000L,(CalibrationPointReference(Some(213),Some(JunctionPointCP)),CalibrationPointReference(None,None)),List(Point(443508.949,7375888.836,0.0), Point(443516.236,7375900.382,0.0)),FrozenLinkInterface,roadwayNumberRightTrackSection,Some(DateTime.now().minusDays(2)),None),
+        LinearLocation(465371,2.0,  "f2e89f78-beff-48c2-82e2-3f57ce345572:2",0.0,8.715,     SideCode.TowardsDigitizing,1698414053000L,(CalibrationPointReference(None,None),CalibrationPointReference(None,None)),List(Point(443498.725,7375709.381,0.0), Point(443499.845,7375718.023,0.0)),FrozenLinkInterface,roadwayNumberRightTrackSection,Some(DateTime.now().minusDays(2)),None),
+        LinearLocation(465381,12.0, "454db012-0d1e-4d2c-b5fb-f124a734ae07:2",0.0,22.514,    SideCode.TowardsDigitizing,1698414053000L,(CalibrationPointReference(None,None),CalibrationPointReference(None,None)),List(Point(443583.051,7376038.167,0.0), Point(443591.418,7376059.053,0.0)),FrozenLinkInterface,roadwayNumberRightTrackSection,Some(DateTime.now().minusDays(2)),None),
+        LinearLocation(466781,4.0,  "bffb8c7d-aac0-4e7c-a115-101e14952a09:2",0.0,113.886,   SideCode.TowardsDigitizing,1698414053000L,(CalibrationPointReference(None,None),CalibrationPointReference(None,None)),List(Point(443818.744,7376495.203,0.0), Point(443864.362,7376599.54,0.0)),FrozenLinkInterface,roadwayNumberCombinedSection,Some(DateTime.now().minusDays(2)),None)
+      )
+
+      linearLocationDAO.create(linearlocations)
+      roadwayDAO.create(roadways)
+      projectDAO.create(Project(projectId, ProjectState.Incomplete, "test", "test", DateTime.now(),"test", DateTime.now(), DateTime.now(),"",Seq(ProjectReservedPart(Sequences.nextViitePrimaryKeySeqValue,roadPart)),Seq(ProjectReservedPart(Sequences.nextViitePrimaryKeySeqValue,roadPart)),None,None,Set(14)))
+      projectReservedPartDAO.reserveRoadPart(projectId, roadPart, "test")
+      projectLinkDAO.create(projectLinks)
+
+      val withoutTerminated = projectLinks.filterNot(_.status == RoadAddressChangeType.Termination)
+      val recalculated = defaultSectionCalculatorStrategy.assignAddrMValues(Seq(), withoutTerminated, Seq())
+      recalculated.length should be (39)
+      val sorted = recalculated.sortBy(pl => (pl.originalStartAddrMValue))
+      val (terminated, nonTerminated) = sorted.partition(_.status == RoadAddressChangeType.Termination)
+      terminated.size should be (2)
```

Comment:
Vastaavasti näitä vastaaville riveille jotkut kommentit, että nää on ne, mitkä pitäis listautua terminoituviin